helix-lock/src/main/java/org/apache/helix/lock/helix/ZKDistributedNonblockingLock.java

```diff
@@ -204,8 +210,20 @@ public ZKDistributedNonblockingLock build() {
       }
 
       // Return a ZKDistributedNonblockingLock instance
-      return new ZKDistributedNonblockingLock(_lockScope.getPath(), _timeout, _lockMsg, _userId,
-          baseDataAccessor);
+      String lockPath = null;
+      if (_lockScope != null) {
+        lockPath = _lockScope.getPath();
+      }
+
+      if (_lockPath != null) {
+        lockPath = _lockPath;
+      }
```

Comment:
I can understand the logic. But it seemed not be a good readable code.

If we think lockPath is necessary:

if (_lockScope != null) {
_lockPath = _lockScope.getPath();
}